test/common/test.js

```diff
@@ -281,47 +270,31 @@ function run(env) {
 					.then(() => nightmare.path())
 					.then(path => {
 						assert.equal(path, '/about');
-
-						return nightmare.evaluate(() => document.querySelector('h1').textContent);
+						return nightmare.title();
 					})
-					.then(header_text => {
-						assert.equal(header_text, 'About this site');
-
+					.then(title => {
+						assert.equal(title, 'About');
 						return nightmare.evaluate(() => window.fulfil({})).wait(100);
 					})
 					.then(() => nightmare.path())
 					.then(path => {
 						assert.equal(path, '/about');
-
-						return nightmare.evaluate(() => document.querySelector('h1').textContent);
+						return nightmare.title();
 					})
-					.then(header_text => {
-						assert.equal(header_text, 'About this site');
-
-						return nightmare.evaluate(() => window.fulfil({})).wait(100);
+					.then(title => {
+						assert.equal(title, 'About');
 					});
 			});
 
 			it('passes entire request object to preload', () => {
 				return nightmare
 					.goto(`${base}/show-url`)
+					.wait(() => window.READY)
 					.evaluate(() => document.querySelector('p').innerHTML)
-					.then(html => {
+					.end().then(html => {
 						assert.equal(html, `URL is /show-url`);
 					});
 			});
-
-			it('calls a delete handler', () => {
-				return nightmare
-					.goto(`${base}/delete-test`)
-					.wait(() => window.READY)
-					.click('.del')
-					.wait(() => window.deleted)
-					.evaluate(() => window.deleted.id)
-					.then(id => {
-						assert.equal(id, 42);
-					});
-			});
 		});
```

Comment:
Awesome stuff!

Is this test supposed to be removed?

You know this better than me, but is it a good idea to have both `package-lock.json` and `yarn.lock` in the same repository?

Reply:
ack! well spotted, thanks — must have got lost when I was trying to merge some stuff. have added it back.

Bloody lockfiles. Removing yarn.lock

Reply:
Can you explain the reason for removing lock files? I always commit mine. (Yarn being the preferred tool in my builds.)

Reply:
Lock files are used to get consistent installs across machines. Having 2 separate lock files in the same repository violates this out of the gate. Having *either* `yarn.lock` or `package-lock.json` in the repository is a good idea, though.

Reply:
Did you mean to say Sapper only supports NPM? That would make more sense. But there's nothing wrong with committing Yarn and NPM lock files—it is just more effort.

Since Yarn has been pretty reliable almost since inception, I barely noticed the release of NPM 5, and just now had to go read up on the difference, including the lock files.

If this ticket is ever closed https://github.com/yarnpkg/yarn/issues/3614 then it may be that Yarn folks won't need to fret about frameworks, etc., that rely solely on NPM, but until then, we should state it more explicitly in the docs that Sapper relies on NPM; use Yarn at your own expense (YMMV).

ALTERNATIVELY, you *can* commit both lock files and explicitly support both, but give the Sapper team a little more work to do. I could easily see y'all doing the first, with a well-worded explanation somewhere.